api/src/services/organizations_v1/create_organization_invitation.py

```diff
@@ -0,0 +1,152 @@
+import logging
+from datetime import timedelta
+from typing import Any
+from uuid import UUID, uuid4
+
+from sqlalchemy import select
+
+from src.adapters import db
+from src.api.route_utils import raise_flask_error
+from src.auth.endpoint_access_util import can_access
+from src.constants.lookup_constants import OrganizationInvitationStatus, Privilege
+from src.db.models.entity_models import LinkOrganizationInvitationToRole, OrganizationInvitation
+from src.db.models.user_models import LinkExternalUser, OrganizationUser, Role, User
+from src.services.organizations_v1.get_organization import get_organization
+from src.services.organizations_v1.update_user_organization_roles import validate_roles
+from src.util import datetime_util
+
+logger = logging.getLogger(__name__)
+
+# Invitation expires after 7 days
+INVITATION_EXPIRY_DAYS = 7
+
+
+def check_duplicate_invitation(
+    db_session: db.Session, organization_id: UUID, invitee_email: str
+) -> None:
+    """Check for existing active invitations for the same email and organization.
+
+    Raises 422 error if an active invitation already exists.
+    Active invitations are those with status 'pending' or 'accepted'.
+    """
+    stmt = (
+        select(OrganizationInvitation)
+        .where(OrganizationInvitation.organization_id == organization_id)
+        .where(OrganizationInvitation.invitee_email == invitee_email.lower())
+    )
+
+    existing_invitation = db_session.execute(stmt).scalar_one_or_none()
+
+    if existing_invitation:
+        if existing_invitation.status in (
+            OrganizationInvitationStatus.PENDING,
+            OrganizationInvitationStatus.ACCEPTED,
+        ):
+            raise_flask_error(
+                422, message=f"An active invitation already exists for {invitee_email}"
+            )
+
+
+def check_user_already_member(
+    db_session: db.Session, organization_id: UUID, invitee_email: str
+) -> None:
+    """Check if the invitee is already a member of the organization.
+
+    Raises 422 error if user is already a member.
+    """
+    # Find user by email through their linked login.gov external user
+
+    stmt = (
+        select(OrganizationUser)
+        .join(User, OrganizationUser.user_id == User.user_id)
+        .join(LinkExternalUser, User.user_id == LinkExternalUser.user_id)
+        .where(OrganizationUser.organization_id == organization_id)
+        .where(LinkExternalUser.email.lower() == invitee_email.lower())
```

Comment:
The emails we store in the DB are already made lowercase, you can exclude it on the DB side to avoid the performance hit that is going to have.